thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -349,10 +355,33 @@ public void serviceAdded(ServiceConfig cfg) throws Exception {
                                 format, maxRequestStringLength, maxRequestContainerLength)));
 
         super.serviceAdded(cfg);
+
+        final DependencyInjector dependencyInjector = cfg.server().config().dependencyInjector();
+        for (ThriftServiceEntry thriftServiceEntry : entries().values()) {
+            for (ThriftFunction thriftFunction : thriftServiceEntry.metadata.functions().values()) {
+                if (!thriftFunction.declaredDecorators().isEmpty()) {
+                    final List<DecoratorAndOrder> decorators = thriftFunction.declaredDecorators();
+                    Function<? super HttpService, ? extends HttpService> decorator = Function.identity();
+                    for (int i = decorators.size() - 1; i >= 0; i--) {
+                        final DecoratorAndOrder d = decorators.get(i);
+                        decorator = decorator.andThen(d.decorator(dependencyInjector));
+                    }
+                    thriftFunction.setDecoratedTHttpService(decorator.apply(this));
+                }
+            }
+        }
     }
 
     @Override
     public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        if (ctx.hasAttr(DECODED_REQUEST)) {
+            final DecodedRequest decodedRequest = ctx.attr(DECODED_REQUEST);
```

Comment:
Micro optimization) We don't have to call both `hasAttr()` and `attr()` because `attr()` returns null if the value is absent. 
```suggestion
        final DecodedRequest decodedRequest = ctx.attr(DECODED_REQUEST);
        if (decodedRequest != null) {
```